pennylane/gradients/gradient_transform.py

```diff
@@ -48,7 +49,51 @@
 ]
 
 
-def gradient_analysis(tape, use_graph=True, grad_fn=None):
+def assert_no_state_returns(measurements):
+    """Check whether a set of measurements contains a measurement process that returns the quantum
+    state and raise an error if this is the case.
+
+    Args:
+        measurements (list[MeasurementProcess]): measurements to analyze
+
+    Currently, the measurement processes that are considered to return the state are
+    ``~.measurements.StateMP``, ``~.measurements.VnEntropyMP``, and ``~.measurements.MutualInfoMP``.
+    """
+    if any(isinstance(m, (StateMP, VnEntropyMP, MutualInfoMP)) for m in measurements):
+        raise ValueError(
+            "Computing the gradient of circuits that return the state is not supported."

```

Comment:
Could we add the transform name here? Also we can mention that's only true for hardware compatible gradient.

Reply:
Thanks!